internal/provider/kubernetes/controller.go

```diff
@@ -1165,20 +1226,30 @@ func (r *gatewayAPIReconciler) watchResources(ctx context.Context, mgr manager.M
 	}
 
 	// Only enqueue EnvoyProxy objects that match this Envoy Gateway's GatewayClass.
+	epPredicates := []predicate.Predicate{
+		predicate.ResourceVersionChangedPredicate{},
+		predicate.NewPredicateFuncs(r.hasManagedClass),
+	}
+	if len(ls) != 0 {
```

Comment:
why is `ls` needed here, can't you just use `r.namespaceLabels` as a way to check

Reply:
good catch. updated